TransformationPlayground/playground_frontend/src/app/page.tsx

```diff
@@ -0,0 +1,206 @@
+        { colspan: 4 }, // Left Column
+        { colspan: 4 }, // Middle Column
+        { colspan: 4 }, // Right Column
+      ]}
+    >
+      {/* User Input Column */}
+      <Container header="User Input Panel">
+        <FormField
+          label="Original JSON Input"
+          description="Paste or edit the JSON blob that represents the Elasticsearch 6.8 Index JSON."
+        >
+          <Textarea
+            value={inputShape}
+            onChange={({ detail }) => setInputShape(detail.value)}
+            placeholder='{"index_name": "my-index", "index_json": { index settings and mappings }}'
+            rows={30}
+          />
+        </FormField>
+      </Container>
+
+      {/* Transformation Column */}
+      <Container header="Transformation Panel">
+        <FormField label="Transformation Settings">
+          <Grid
+            gridDefinition={[
+              { colspan: 6 }, // Source Version Dropdown
+              { colspan: 6 }, // Target Version Dropdown
+              { colspan: 6 }, // Transform Type Dropdown
+              { colspan: 6 }, // Transform Language Dropdown
+            ]}
+          >
+            <Select
+              selectedOption={sourceVersion}
+              onChange={({ detail }) => setSourceVersion(detail.selectedOption!)} 
+              options={sourceVersionOptions}
+              placeholder="Select Source Version"
+            />
+            <Select
+              selectedOption={targetVersion}
+              onChange={({ detail }) => setTargetVersion(detail.selectedOption!)}
+              options={targetVersionOptions}
+              placeholder="Select Target Version"
+            />
+            <Select
+              selectedOption={transformType}
+              onChange={({ detail }) => setTransformType(detail.selectedOption!)}
+              options={transformTypeOptions}
+              placeholder="Select Transform Type"
+            />
+            <Select
+              selectedOption={transformLanguage}
+              onChange={({ detail }) => setTransformLanguage(detail.selectedOption!)}
+              options={transformLanguageOptions}
+              placeholder="Select Language"
+            />
+          </Grid>
+        </FormField>
+        <br />
+        <Button onClick={handleGetRecommendation}>
+          {isLoading ? <Spinner/> : "Get GenAI Recommendation"}
+        </Button>
+        <Button onClick={() => {
+          setTransformLogic("");
+          setOutputShape("");
+          setValidationReport("");
+        }}>
+          Clear Transform
+        </Button>
+        <FormField label="Generated Transformation Logic">
+          <Textarea
+            value={transformLogic}
+            readOnly
+            placeholder="The Python code for the transformation will appear here."
+            rows={30}
+          />
+        </FormField>
+      </Container>
+
+      {/* Testing/Output Column */}
+      <Container header="Testing & Output Panel">
```

Comment:
How do you cycle back and forth between source / transformation logic / output?

Reply:
I haven't implemented this all yet (obviously), but here's the user journey I'm envisioning:
* The user loads in their input shapes.  Currently, that's just copy/pasting JSON into a text field, but soon I'd like to be able to load the contents of a snapshot into the Playground via a configurable process.  Imagine a dialogue where you select the S3 location of your snapshot, something extracts the templates, indices, and a configurable number of documents (that's a whole workstream right there), and presents them in the left panel so you can see everything in your snapshot, then click on specific items to see their JSON, etc.  Even once we've added the ability to read from snapshots, we'll want to leave open the possibility of adding manual input shapes.  Example - the user doesn't have access to a snapshot of their production cluster but they do have access to the settings JSON and a few documents, which they can manually enter as shapes into the Playground for testing.
* The user then uses the dropdowns to select what type of transformation they are performing.  The "transform type" (Index vs. template vs. documents) dropdown will disappear, because it will be obvious from context (the user selected an index on the input panel, so...).
* Once that is selected, we will have a pre-canned transform (we'll hand-craft and save, no GenAI) that is supplied based on those selections that will serve as a default.  The default will be populated into the UI and pinned against that shape in the backend.
* The user can then hit a "test" button to see how that default transform will work against their input shape (e.g. we kick off the validation process).
* If the user wants to include functional tests into validation (e.g. actually creating/deleting indices against a real test cluster), they can go through a dialogue to set up and test a connection to their target cluster.  Currently, that process is just "paste a URL into a text field and assume there's no auth to worry about").
* If the user likes the results of the default, pre-canned transform and it passes validation - great!  No further work needed.  If not, they can either modify the transform directly in the UI and run the validation process again, or they can ask the GenAI assistant to modify the existing transform in some way (not currently implemented, but expect to be easy).  Either way, the new transform is then pinned against the input shape in the backend so that we know there's something custom going on.
* The user goes through this process until everything transforming as they desire, then they can either hit a button to "deploy" their transformed data/metadata to the target cluster (think for metadata migrations which are inherently "low scale") or "bundle" the transformations for inclusion into the Migrations Assistant backfill/replay processes (which are anything but "low scale").  For the "deploy" button, this is basically just skipping the cleanup step of the validation process we've already performed (e.g. take the transformed output and PUT it against the target, but don't delete it afterwards).  For bundling, this is taking the stuff we have in the server-side DB, packaging it into a tarball/zip/whatever, and sticking it somewhere (S3?) so that the backfill and replayer processes can pick it up and load the transform objects.